lib/spack/spack/build_environment.py

```diff
@@ -250,8 +254,17 @@ def set_build_environment_variables(pkg, env):
         env.prepend_path('PATH', item)
     env.set_path(SPACK_ENV_PATH, env_paths)
 
+    dep_prefixes = []
+    for d in pkg.spec.traverse(root=False):
+        # Set the dynamic linker to the path specified by the dependency
+        dep_pkg = d.package_class
+        if dep_pkg.dynamic_linker:
```

Comment:
Should we fail if something depends on two packages that provide the `dynamic_linker` property? Or should we silently accept only the latest one?